monix-catnap/shared/src/test/scala/monix/catnap/ConcurrentChannelSuite.scala

```diff
@@ -113,6 +113,41 @@ abstract class BaseConcurrentChannelSuite[S <: Scheduler] extends TestSuite[S] {
     } yield r
   }
 
+  testIO("consumer queue is empty after pulling all messages", times = repeatForFastTests) { implicit ec =>
+    for {
+      chan <- ConcurrentChannel[IO].withConfig[Int, Int](boundedConfig)
+      consume = chan.consume.use { consumer =>
+        for {
+          _ <- consumer.pull
+        } yield {
+          assertEquals(consumer.isEmpty.unsafeRunSync(), true)
```

Comment:
That's actually an odd test. I'd expect `isEmpty` to be false, since you are also able to `pull` a halt event (`Left`).
You might also want to synchronize your producer and consumer with a `Deferred` to know exactly when pushing/halting has completed.

Reply:
i had a lot of doubts when writing this test, i'll try implementing it another way